res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+    if (!trackLoaded) {
+        return;
+    }
+
+    if (value && !cue) {
+        // play illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.playFromCuePoint = true;
+        return;
+    }
+    if (!value && !cue) {
+        // pause illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, false);
+        PioneerDDJSX.playFromCuePoint = false;
+        return;
+    }
+    if (value && cue) {
+        // stop and go to cue point illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.cuePressed[deck] = false;
+        PioneerDDJSX.playFromCuePoint = true;
+    }
+    if (!value && cue) {
+        if (PioneerDDJSX.playFromCuePoint) {
+            // play from cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.playFromCuePoint = false;
+        } else {
+            // set cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        }
+        PioneerDDJSX.cuePressed[deck] = false;
+    }
+};
+
+PioneerDDJSX.playStutterIllumination = function(value, group, control) {
+    if (value) {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+    } else {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+    }
+};
+
+PioneerDDJSX.cueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.cue, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftCue, value);
+};
+
+PioneerDDJSX.loadLed = function(value, group, control) {
+    var deck = PioneerDDJSX.channelGroups[group];
+    PioneerDDJSX.generalLedControl(PioneerDDJSX.nonPadLeds["loadDeck" + (deck + 1)], value);
+    PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.loadedDeck, value);
+};
+
+PioneerDDJSX.reverseLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.censor, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDD
```

Comment:
I accidentally made the comment on the wrong line of code. I meant to say you can remove your `PioneerDDJSX.softTakeoverEmulation` function.

Yes, you do need `engine.softTakeover` for both `engine.setValue` and `engine.setParameter`. In Mixxx 2.0, there was a bug that made `engine.softTakeover` not work when `engine.setParameter` was used, but is fixed in the master branch and will be in Mixxx 2.1. Note you need to call `engine.softTakeoverIgnoreNextValue` when switching what the potentiometer controls, as documented on the wiki.